Lombiq.Tests.UI/SecurityScanning/YamlDocumentExtensions.cs

```diff
@@ -0,0 +1,425 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using YamlDotNet.RepresentationModel;
+
+namespace Lombiq.Tests.UI.SecurityScanning;
+
+public static class YamlDocumentExtensions
+{
+    /// <summary>
+    /// Sets the start URL under the app where to start the scan from in the current context of the ZAP Automation
+    /// Framework plan.
+    /// </summary>
+    /// <param name="startUri">The absolute <see cref="Uri"/> to start the scan from.</param>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the ZAP Automation Framework plan contains more than a single URL in the "urls" section.
+    /// </exception>
+    public static YamlDocument SetStartUrl(this YamlDocument yamlDocument, Uri startUri)
+    {
+        // Setting includePaths in the context is not necessary because by default everything under "urls" will be
+        // scanned.
+
+        var urls = yamlDocument.GetUrls();
+        urls.Children.Clear();
+        urls.Add(startUri.ToString());
```

Comment:
Adding the start URI at the first position (without clearing the rest) would be more intuitive. This way you'd have to be aware that calling SetStartUrl and AddUrl in a different order would give a different result. Sorry, I wanted to suggest this in advance in my earlier comment, but forgot.

Reply:
Clearing is necessary if we want to allow the usage of AF plans that you can manually execute from ZAP too. Due to the general recommendation that you should start with ZAP from the UI, including building custom AF plans, that's a common use case.

For this, you need the URL statically added, and it'll be the one you run the app under (like with IIS Express). This can't be the same as the one used for UI testing, since that'll use a random port. So, we can't include both of them, and you can't pre-add any other URL to the plan for the same reason either.

We could require an empty URLs section instead and throw an exception, but that's less convenient.

Reply:
Oh, I see. With these abstraction layers it comes off a bit weird, but it's necessary, OK!